### DISCUSSION TITLE
How to create a delayed clock?

### DISCUSSION THREAD

**Discussion by matheusinacio7** [hneemann/Digital#970] [Q&A] [answered]
I'm learning about electronics now; reading the introduction book "But How Do It Know?" and reproducing the circuits using Digital (thanks a lot for the software!)

There's one problem though: the book mentions delaying a clock by one quarter of a cycle to have a "clk d" output; we can then AND clk + clk d to have a clock that is on for a shorter period of time and OR clk + clk d to have a clock that is on for a longer period of time - the author calls them "clk e" and "clk s" because clk e + clk s sets a good tempo for setting things down the bus

I haven't been able to reproduce this in Digital; the "delay" component doesn't seem to be delaying at all :thinking: 
![image](https://user-images.githubusercontent.com/52154225/175268408-1a6735e6-449d-4b02-97be-b5e61b8699d5.png)
I also know that in Digital the wires have no delay - so no matter the length of the wire, the data goes down instantly

Is it possible to simulate the "delayed clock by 1 quarter of a cycle" behavior?

**Comment by hneemann** [comment, marked as answer]
You must generate the clock signals with a faster running clock:

![clock](https://user-images.githubusercontent.com/7190563/175280656-e4e001c7-d5fa-4adc-806b-9eb3801fb441.png)

The "Delay" component delays a signal by just one gate propagation time.

**Reply by matheusinacio7** [reply]
Awesome! Makes a lot of sense; I didn't know about this D flip flop; it's simply a flip-flop that also outputs the reversed value, yeah?

Thanks so much for the quick answer!
